packages/firestore/src/remote/watch_change.ts

```diff
@@ -188,30 +192,84 @@ export class WatchChangeAggregator {
    * so to know if a target is active, there must be no pending acks we're
    * waiting for and it must be in the current list of targets that the client
    * cares about.
+   */
+  protected queryDataForActiveTarget(targetId: TargetId): QueryData | null {
+    const queryData = this.listenTargets[targetId];
+    return queryData &&
+      !objUtils.contains(this.pendingTargetResponses, targetId)
+      ? queryData
+      : null;
+  }
+
+  /**
+   * Defers to queryForActiveTarget to determine if the given targetId
+   * corresponds to an active target.
    *
    * This method is visible for testing.
    */
   protected isActiveTarget(targetId: TargetId): boolean {
-    return (
-      !objUtils.contains(this.pendingTargetResponses, targetId) &&
-      objUtils.contains(this.listenTargets, targetId)
-    );
+    return this.queryDataForActiveTarget(targetId) !== null;
+  }
+
+  private updateLimboDocuments(
```

Comment:
You wrote such a nice comment in the iOS implementation. It'd be awesome to see it here as well.

Reply:
Oops, forgot to copy that. Fixed.